### PR TITLE
PWX-32340: Adding Failback workflow tests with new action workflow with different schedule policies for reverse migrationschedule.

### PR DESCRIPTION
**Pull request by diptiranjanpx**
Signed-Off-By: Diptiranjan

**What type of PR is this?**
integration-test

**What this PR does / why we need it**:
Added failback tests with reverse migration schedule created with different schedule policies.

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
no

**Does this change need to be cherry-picked to a release branch?**:
yes, 24.2.0

**Test**
```
time="2024-04-22T18:35:42Z" level=info msg="#Result: PASS "
time="2024-04-22T18:35:42Z" level=info msg=------------------------
time="2024-04-22T18:35:42Z" level=warning msg="Skipping testrail update for this case, testID: 296302, testrun: 0"
2024/04/22 18:35:42 WARN: Skipping testrail update for this case, testID: 296302, testrun: 0
--- PASS: TestDRActions (3632.30s)
    --- PASS: TestDRActions/testDRActionFailbackIntervalScheduleTest (935.24s)
    --- PASS: TestDRActions/testDRActionFailbackDailyScheduleTest (925.29s)
    --- PASS: TestDRActions/testDRActionFailbackWeeklyScheduleTest (850.51s)
    --- PASS: TestDRActions/testDRActionFailbackMonthlyScheduleTest (912.57s)
PASS
time="2024-04-22T18:35:43Z" level=info msg="TestSetId 621916 update successfully"
time="2024-04-22T18:35:43Z" level=info msg="Dashboard URL : https://aetos.pwx.purestorage.com/resultSet/testSetID/621916"

DONE 5 tests in 3731.806s
Tests passed
```

### CODE REVIEW

File: test/integration_test/migration_dr_actions_test.go [hunk 1]
```diff
@@ -498,6 +494,295 @@ func testDRActionFailoverWithMigrationRunningTest(t *testing.T) {
...
+	schedulePolicyDestArgs["migrate-every-month-dr"] = map[string]string{"policy-type": "Monthly"}
+	failBackWithMigrationSchedulesWithDifferentPolicies(t, instanceID, mysqlApp, mysqlNamespace, schedulePolicyDestArgs)
+
+	// If we are here then the test has passed
+	testResult = testResultPass
+	log.InfoD("Test status at end of %s test: %s", t.Name(), testResult)
+}
+
+func failBackWithMigrationSchedulesWithDifferentPolicies(t *testing.T, instanceID string, appName string, appNamespace string, reverseSchedulePolicyArgs map[string]map[string]string) {
+	// Schedule multiple apps
```
Comment (olavangad-px) [inline test/integration_test/migration_dr_actions_test.go:609]: nit.. this comment should be removed

File: test/integration_test/migration_dr_actions_test.go [hunk 2]
```diff
@@ -498,6 +494,295 @@ func testDRActionFailoverWithMigrationRunningTest(t *testing.T) {
...
+	Dash.VerifyFatal(t, *sourceDeployments.Items[0].Spec.Replicas, 0, fmt.Sprintf("Expected 0 replica in source in deployment in %s namespace", appNamespace))
+
+	// Failback the application
+	err = setDestinationKubeConfig()
+	log.FailOnError(t, err, "failed to set kubeconfig to destination cluster: %v", err)
+	err = setMockTime(nil)
+	log.FailOnError(t, err, "Error resetting mock time")
+
+	// migrationScheduleArgs is a map of migrationScheduleName : {{flag1:value1,flag2:value2,....}}
+	reverserMigrationScheduleArgs := make(map[string]map[string]string)
```
Comment (olavangad-px) [inline test/integration_test/migration_dr_actions_test.go:702]: nit.. typo in the name `reverseMigrationScheduleArgs`

File: test/integration_test/migration_dr_actions_test.go [hunk 3]
```diff
@@ -515,3 +800,83 @@ func waitTillActionComplete(t *testing.T, actionType storkv1.ActionType, actionN
...
+	default:
+		log.FailOnError(t, fmt.Errorf("invalid policy type %s", policyType), "Invalid policy type")
+	}
+	if authTokenConfigMap != "" {
+		err := addSecurityAnnotation(schedPolicy)
+		log.FailOnError(t, err, "error updating security annotations for schedule policy %s of type %v", policyName, policyType)
+	}
+
+	_, err := storkops.Instance().CreateSchedulePolicy(schedPolicy)
+	log.FailOnError(t, err, "error creating monthly schedule policy")
```
Comment (olavangad-px) [inline test/integration_test/migration_dr_actions_test.go:878]: nit.. log statement is hardcoded with monthly type.

**Review by olavangad-px — APPROVED** [review]
LGTM

File: test/integration_test/migration_dr_actions_test.go (hunk 1 above)
Comment (diptiranjanpx) [reply test/integration_test/migration_dr_actions_test.go]: Done

File: test/integration_test/migration_dr_actions_test.go (hunk 2 above)
Comment (diptiranjanpx) [reply test/integration_test/migration_dr_actions_test.go]: Done

File: test/integration_test/migration_dr_actions_test.go (hunk 3 above)
Comment (diptiranjanpx) [reply test/integration_test/migration_dr_actions_test.go]: Done

**Review by olavangad-px — APPROVED** [review]
LGTM.
Please attach the job link to the PR as well if one is created already

**Comment by diptiranjanpx** [comment]
> LGTM. Please attach the job link to the PR as well if one is created already

https://jenkins.pwx.dev.purestorage.com/job/Users/job/dipti/job/dipti-new-dr-workflow/10/

**Pull request merged**